### PR TITLE
dayon: init at 11.0.7

### PR DESCRIPTION
**Pull request by fgaz**
###### Description of changes

An easy to use, cross-platform remote desktop assistance solution

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/remote/dayon/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    owner = "RetGal";
+    repo = "dayon";
+    rev = "v${version}";
+    hash = "sha256-3TbJVM5po4aUAOsY7JJs/b5tUzH3WGnca/H83IeMQ2s=";
+  };
+
+  postPatch = ''
+    substituteInPlace resources/deb/dayon_assisted.desktop resources/deb/dayon_assistant.desktop \
+      --replace "Exec=/usr/bin" "Exec=$out/bin" \
+      --replace "Icon=/usr/bin/dayon/dayon.png" "Icon=dayon.png"
```
Comment (davidak) [inline pkgs/applications/networking/remote/dayon/default.nix:25]: You could propose a patch to upstream to make it more portable for everyone.

**Review by davidak — APPROVED** [review]
Change looks good

Not tested

Thank you! :sparkles:

File: pkgs/applications/networking/remote/dayon/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/applications/networking/remote/dayon/default.nix:25]: You mean using basenames for both?

Comment (davidak) [reply pkgs/applications/networking/remote/dayon/default.nix:25]: Whatever helps it to work not only in `/usr/bin`. I don't know the solution.

Comment (fgaz) [reply pkgs/applications/networking/remote/dayon/default.nix]: I proposed a patch upstream https://github.com/RetGal/Dayon/pull/66

**Pull request merged**